src/stdatamodels/fits_support.py

```diff
@@ -497,14 +497,21 @@ def to_fits(tree, schema, hdulist=None):
     if _ASDF_EXTENSION_NAME in hdulist:
         del hdulist[_ASDF_EXTENSION_NAME]
 
-    hdulist.append(_create_asdf_hdu(tree))
+    # Do not pass "extra_fits" to _create_asdf_hdu
+    # as we've already saved the contents above in _save_extra_fits
+    hdulist.append(_create_asdf_hdu({k: v for k, v in tree.items() if k != 'extra_fits'}))
```

Comment:
The contents of `extra_fits` are saved a few lines prior:
https://github.com/spacetelescope/stdatamodels/blob/6b62413268edace5295e82e62911875e65b99629/src/stdatamodels/fits_support.py#L491
This line strips the contents of `extra_fits` prior to saving. Note that the contents are already stripped on read:
https://github.com/spacetelescope/stdatamodels/blob/6b62413268edace5295e82e62911875e65b99629/src/stdatamodels/fits_support.py#L625-L628

Reply:
I'm removing this as testing this with other files revealed an odd behavior of `extra_fits` where it sometimes does not load. Until I understand why it sometimes does not load I'd rather leave the duplicate data in place.